### ISSUE TITLE
Specify how exporters should deal with invalid instrumentation library names

### ISSUE THREAD

**Issue by Oberon00** [open-telemetry/opentelemetry-specification#1554]
Since #1534 / #1472, we pass on invalid (i.e. null/empty) instrumentation library names to the exporter. We should specify how specific exporters should deal with it. In my opinion, the goal should be to alert users that they are making a mistake when they decline (not forget: the null/empty string has to be explicitly passed) to specify an instrumentation library name. So I would suggest:

* OTLP: Probably pass it on as-is, but add SHOULD/MUST-requirement for consumers (e.g. collector when translating to other protocol) to alert users somehow.
* Jaeger/Zipkin: Replace with `"<INSTRUMENTATION LIBRARY NAME MISSING>"` or similarly screaming string.

**Comment by carlosalberto** [comment]
@yurishkuro @anuraaga do you have any opinion about Jaeger and Zipkin handling of this, respectively?

**Comment by yurishkuro** [comment]
Jaeger has no requirements on instrumentation library name, it only requires service name. I assume the same holds for Zipkin. So I am not clear what value a long screaming string would provide. I would simply omit setting this on the spans.

**Comment by Oberon00** [comment]
> Jaeger has no requirements on instrumentation library name,

But OpenTelemetry has. And in  #1534 / #1472 it was decided that the backend is responsible for showing the error message to the user. Since I don't think Jaeger/Zipkin will add special support for the OTel library names, the last point in the chain where we can add the error message is the exporter.

> So I am not clear what value a long screaming string would provide.

It will urge users to put in place an instrumentation library name, therefore stop calling the OpenTelemetry getTracer API with invalid arguments.

**Comment by yurishkuro** [comment]
> backend is responsible for showing the error message to the user

I don't think this is stated anywhere as a requirement. Most users won't even care about this value, I think the main persona who cares is someone who's investigating instrumentation issues, and for them the absence of this value tells as much as a long screaming string.
